staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -676,18 +678,15 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 		structuralSchemas[v.Name] = s
 	}
 
-	openAPIModels, err := buildOpenAPIModelsForApply(r.staticOpenAPISpec, crd)
+	// CRD group-versions each get their own type converter.
+	// Fetch the ObjectMeta and TypeMeta schemas to allow this
+	typeConverter, err := buildOpenAPIModelsForApply(r.staticTypeConverter, crd)
 	if err != nil {
 		utilruntime.HandleError(fmt.Errorf("error building openapi models for %s: %v", crd.Name, err))
-		openAPIModels = nil
-	}
 
-	var typeConverter fieldmanager.TypeConverter = fieldmanager.DeducedTypeConverter{}
-	if openAPIModels != nil {
-		typeConverter, err = fieldmanager.NewTypeConverter(openAPIModels, crd.Spec.PreserveUnknownFields)
-		if err != nil {
-			return nil, err
-		}
+		// Fall back to deduced type converter in case of failure to setup SSA for
+		// CRDs
+		typeConverter = fieldmanager.DeducedTypeConverter{}
```

Comment:
At the very least need the boolean option since some CRDs specify Spec.PreserveUnknownFIelds which predates x-kubernetes-preserve-unknown-fields. We could the extension from the bool value?